### PR TITLE
Docstrings describing inputs to writer functions

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
The cookiecutter writer functions do not describe what inputs they expect adequately. This PR attempts to improve that.

In particular, it is confusing because the input `data` in function `write_single_image` is not the same type of object as the input `data` in function `write_multiple`.

I've used numpy style docstring formatting here, mostly because I didn't see enough other docstrings to know what docstring format I should use for this project. We can change it, of course.

### CODE REVIEW

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 1]
```diff
@@ -16,7 +16,17 @@
 
 
 def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
-    """Writes a single image layer"""
+    """Writes a single image layer.
+    
+    Parameters
+    ----------
+    path : str
+    data : The layer data
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:24]: `meta` should be also documented. As it contains all layer attributes different from the data.

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 2]
```diff
@@ -25,7 +35,21 @@ def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
...
-    """Writes multiple layers of different types."""
+    """Writes multiple layers of different types.
+    
+    Parameters
+    ----------
+    path : str
+    data : A layer tuple.
+           Tuple contains three elements: (data, kwargs, layer_type)
+           `data` is the layer data
+           `kwargs` is a dictionary of keyword arguments
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:46]: Should be updated to meta to have the same names, as is same as meta in previous function.

**Review by Czaki — CHANGES_REQUESTED** [review]

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 3]
```diff
@@ -25,7 +40,23 @@ def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
 
 
 def write_multiple(path: str, data: List[FullLayerData]) -> List[str]:
-    """Writes multiple layers of different types."""
+    """Writes multiple layers of different types.
+    
+    Parameters
+    ----------
+    path : str
+           A string path indicating where to save the image file.
```
Comment (jni) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:48]: Indent is a bit off here? should be 4 spaces.

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 4]
```diff
@@ -25,7 +40,23 @@ def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
...
+    data : A layer tuple.
```
Comment (jni) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:49]: list of!

**Review by jni — CHANGES_REQUESTED** [review]
@GenevieveBuckley thanks for this, it's very handy! 😅 I've requested two minor changes, one of which is important (the type of `data` is incorrect for `write_multiple`)

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py (hunk 4 above)
Comment (GenevieveBuckley) [reply {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py]: Aaargh
Thanks! This is why we need good doctrings 😆

**Comment by GenevieveBuckley** [comment]
Requested changes addressed. Should be good to merge when the CI tests pass

**Review by Czaki — APPROVED** [review]

**Comment by GenevieveBuckley** [comment]
@jni when you get a sec could you merge this? Thanks

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 5]
```diff
@@ -25,7 +40,23 @@ def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
 
 
 def write_multiple(path: str, data: List[FullLayerData]) -> List[str]:
-    """Writes multiple layers of different types."""
+    """Writes multiple layers of different types.
+    
+    Parameters
+    ----------
+    path : str
+        A string path indicating where to save the image file.
```
Comment (jni) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:48]:
```suggestion
        A string path indicating where to save the data file(s).
```

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 6]
```diff
@@ -25,7 +40,23 @@ def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
...
+    data : A list of layer tuples.
+        Tuples contain three elements: (data, meta, layer_type)
+         data` is the layer data
+        `meta` is a dictionary containing all other metadata attributes
+        from the napari layer (excluding the `.data` layer attribute).
+        `layer_type` is a string, eg: "image", "labels", "surface", etc.
+
+    Returns
+    -------
+    [path] : A list containing multiple string paths to the saved files.
```
Comment (jni) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:58]:
```suggestion
    [path] : A list containing (potentially multiple) string paths to the saved file(s).
```

**Review by jni — CHANGES_REQUESTED** [review]
@GenevieveBuckley sorry for the delay — could you have a look at my suggestions? Since `multiple_` could save non-image layers, I changed the language of that docstring slightly.

**Comment by GenevieveBuckley** [comment]
Done - good suggestions @jni

File: {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py [hunk 7]
```diff
@@ -25,7 +40,23 @@ def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
...
-    """Writes multiple layers of different types."""
+    """Writes multiple layers of different types.
+    
+    Parameters
+    ----------
+    path : str
+        A string path indicating where to save the data file(s).
+    data : A list of layer tuples.
+        Tuples contain three elements: (data, meta, layer_type)
+         data` is the layer data
```
Comment (jni) [inline {{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py:51]:
```suggestion
        `data` is the layer data
```

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Hmmm I thought #150 would fix the build but it didn't... If anyone has ideas about that (@Czaki @aganders3 @tlambert03?) I would appreciate them! The [build logs](https://github.com/napari/cookiecutter-napari-plugin/actions/runs/4614369575/jobs/8157264751?pr=154#step:6:1869) aren't very informative:

```pytb
>               raise CalledProcessError(retcode, process.args,
[1965](https://github.com/napari/cookiecutter-napari-plugin/actions/runs/4614369575/jobs/8157264751?pr=154#step:6:1968)
                                         output=stdout, stderr=stderr)
[1966](https://github.com/napari/cookiecutter-napari-plugin/actions/runs/4614369575/jobs/8157264751?pr=154#step:6:1969)
E               subprocess.CalledProcessError: Command '['pre-commit', 'run', '--all']' returned non-zero exit status 1.
```

from `test_pre_commit_validity` in all configurations...

**Comment by Czaki** [comment]
@jni the #156 is fixing pre-commit related things

**Pull request merged**